sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -633,10 +633,8 @@ void gcoap_register_listener(gcoap_listener_t *listener)
 }
 
 int gcoap_req_init(coap_pkt_t *pdu, uint8_t *buf, size_t len,
-                   unsigned code, const char *path)
+                   unsigned code, const char *path, bool proxied)
```

Comment:
Are you sure we want to specify use of a proxy here? The major issue I see is that the option number for Proxy-Uri is 35.  If we add this option here, the user will be unable to specify use of an option with a lesser number. These lesser numbers include the large number of options, including Content-Format.

Why not just require the user to specify use of a forward proxy by with `coap_opt_add_string()` themselves? This allows them to use either Proxy-Uri or Proxy-Scheme.

Reply:
hmm, this would mean:
* no changes in gcoap (maybe a little bit on the documentation on how to achieve proxied requests)
* put more logic into the gcoap example

or did I misunderstand your comment?

Reply:
Yes, right on both points -- and point to the example from the documentation.

Reply:
You also could add an inline convenience function, `coap_opt_add_proxy_uri()`, similar to `coap_opt_put_location_path()`.

Reply:
> You also could add an inline convenience function, `coap_opt_add_proxy_uri()`, similar to `coap_opt_put_location_path()`.

I think that approach makes sense in this case. +1